src/cljs/main/broadfcui/page/workspace/notebooks/notebooks_table.cljs

```diff
@@ -0,0 +1,443 @@
+           view-error-link (links/create-internal {:data-test-id clusterNameStatusId
+                                                   :style {:textDecoration "none" :color (:button-primary style/colors)}
+                                                   :onClick #(swap! state assoc :show-error-dialog? true :errored-cluster-to-show cluster)}
+                                                  "View error")]
+       [:div {:key (when cluster-map (str (gensym))) ;this makes the spinners sync
+              :style {:height table-style/table-icon-size}}
+        (case (:status cluster)
+          "Error" [:div {:style {:display "inline-flex"}} (moncommon/render-failure-icon) view-error-link]
+          ("Creating" "Updating" "Deleting" "Starting" "Stopping") spinner-icon
+          "Running" stop-cluster-link
+          "Stopped" start-cluster-link
+          (moncommon/render-unknown-icon))
+        (when-not (= (:status cluster) "Error")
+          [:span {:data-test-id clusterNameStatusId} (:status cluster)])]))
+
+   :-choose-cluster
+   (fn [{:keys [state props]} selected-cluster]
+     (let [{:keys [choose-notebook]} @state]
+       ((:choose-cluster props) choose-notebook selected-cluster)))})
+
+(react/defc NotebooksContainer
+  {:refresh
+   (fn [{:keys [this]}]
+     (this :-get-pet-token)
+     (this :-get-clusters-list)
+     (this :-schedule-cookie-refresh))
+
+   :render
+   (fn [{:keys [props state this]}]
+     (let [{:keys [server-response cluster-map show-create-dialog?]} @state
+           {:keys [notebooks server-error]} server-response]
+       [:div {:display "inline-flex"}
+        (when show-create-dialog?
+          [NotebookCreator (assoc props :dismiss #(swap! state dissoc :show-create-dialog?)
+                                        :pet-token (:pet-token @state)
+                                        :notebooks notebooks
+                                        :refresh-notebooks #(this :-refresh-notebooks))])
+        [:div {} [:span {:data-test-id "notebooks-title" :style {:fontSize "125%" :fontWeight 500 :paddingBottom 10 :marginLeft 10}} "Notebooks"]]
+        [:div {:style {:margin 10 :fontSize "88%"}}
+         "Create a Jupyter notebook and launch an interactive analysis environment based on Spark and Hail.
+         See online documentation " [:a {:href (config/user-notebooks-guide-url) :target "_blank"} "here" icons/external-link-icon]]
+        [comps/ErrorViewer {:data-test-id "notebooks-error" :error server-error}]
+        (if notebooks
+          [NotebooksTable
+           (assoc props
+             :toolbar-items [flex/spring [buttons/Button {:data-test-id "create-modal-button"
+                                                          :text "Create Notebook..." :style {:marginRight 1}
+                                                          :onClick #(swap! state assoc :show-create-dialog? true)}]
+                             ; TODO: upload notebook not yet implemented
```

Comment:
Noting another TODO -- I realized late that we'd need "upload notebook" in addition to "create notebook".

Reply:
This is implemented now